### PR TITLE
Fix documentation for DeltaTable.forPath

### PR DESCRIPTION
**Pull request by yikf**
## Description
fix https://github.com/delta-io/delta/issues/1272
## How was this patch tested?
doc update only

## Does this PR introduce _any_ user-facing changes?
No, just internal based-code comment updated

### CODE REVIEW

File: core/src/main/scala/io/delta/tables/DeltaTable.scala [hunk 1]
```diff
@@ -618,7 +618,9 @@ object DeltaTable {
   }
 
   /**
-   * Create a DeltaTable for the data at the given `path`.
+   * Instantiate a DeltaTable representing the data at the given `path`, If the given path is
+   * invalid (i.e. either no table exists or an existing table is not a delta table), it throws
+   * a `notADeltaTableException` Exception.
```
Comment (vkorukanti) [inline core/src/main/scala/io/delta/tables/DeltaTable.scala:623]: Instantiate a [[DeltaTable]] object representing the data at the given `path`, If the given path is
invalid (i.e. either no table exists or an existing table is not a Delta table), it throws a 'not a Delta table`
error.

**Review by vkorukanti — CHANGES_REQUESTED** [review]
lgtm, minor changes suggested.

File: core/src/main/scala/io/delta/tables/DeltaTable.scala (hunk 1 above)
Comment (yikf) [reply core/src/main/scala/io/delta/tables/DeltaTable.scala]: updated

**Review by vkorukanti — APPROVED** [review]

**Comment by zsxwing** [comment]
@yikf Could you also submit a PR to fix the doc for [DeltaTable.forName](https://github.com/delta-io/delta/blob/696c342400bad459258002c6b4bc347590473cc0/core/src/main/scala/io/delta/tables/DeltaTable.scala#L662) and [Python APIs](https://github.com/delta-io/delta/blob/56e1b9b417e8815cd7003eaf1796988c670b8b0d/python/delta/tables.py#L334)? Please don't push new commits to this PR as we are merging it. Thanks!

**Pull request closed without merging**